File: app/src/test/java/com/benoitquenaudon/tvfoot/red/app/injection/module/ServiceMockModule.java
package com.benoitquenaudon.tvfoot.red.app.injection.module;

import dagger.Module;
import dagger.Provides;
import com.benoitquenaudon.tvfoot.red.api.TvfootService;
import com.benoitquenaudon.tvfoot.red.testutil.FakeTvfootService;
import com.benoitquenaudon.tvfoot.red.testutil.Fixture;
import javax.inject.Singleton;

@Module public class ServiceMockModule {
  @Provides @Singleton TvfootService provideMatchService(Fixture fixture) {
    return new FakeTvfootService(fixture);
  }
}

File: app/src/test/java/com/benoitquenaudon/tvfoot/red/app/injection/module/ServiceMockModule.kt
package com.benoitquenaudon.tvfoot.red.app.injection.module

import com.benoitquenaudon.tvfoot.red.api.TvfootService
import com.benoitquenaudon.tvfoot.red.testutil.FakeTvfootService
import com.benoitquenaudon.tvfoot.red.testutil.Fixture
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module object ServiceMockModule {
  @JvmStatic @Provides @Singleton fun provideMatchService(fixture: Fixture): TvfootService {
    return FakeTvfootService(fixture)
  }
}